### ISSUE TITLE
issue while cordova build andoird getting these error

### ISSUE THREAD

**Issue by satsathish** [phonegap/phonegap-plugin-push#2250]
Subproject Path: CordovaLib
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_1cwws06hwd03bm35gf7tvrzrn.run(D:\notification\platforms\android\build.gradle:142)
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
Incremental java compilation is an incubating feature.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
:preBuild UP-TO-DATE
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:preDebugAndroidTestBuild UP-TO-DATE
:preReleaseBuild UP-TO-DATE
:CordovaLib:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:checkDebugManifest
:CordovaLib:prepareDebugDependencies
:CordovaLib:compileDebugAidl UP-TO-DATE
:CordovaLib:compileDebugNdk UP-TO-DATE
:CordovaLib:compileLint UP-TO-DATE
:CordovaLib:copyDebugLint UP-TO-DATE
:CordovaLib:mergeDebugShaders UP-TO-DATE
:CordovaLib:compileDebugShaders UP-TO-DATE
:CordovaLib:generateDebugAssets UP-TO-DATE
:CordovaLib:mergeDebugAssets UP-TO-DATE
:CordovaLib:mergeDebugProguardFiles UP-TO-DATE
:CordovaLib:packageDebugRenderscript UP-TO-DATE
:CordovaLib:compileDebugRenderscript UP-TO-DATE
:CordovaLib:generateDebugResValues UP-TO-DATE
:CordovaLib:generateDebugResources UP-TO-DATE
:CordovaLib:packageDebugResources UP-TO-DATE
:CordovaLib:processDebugManifest UP-TO-DATE
:CordovaLib:generateDebugBuildConfig UP-TO-DATE
:CordovaLib:processDebugResources UP-TO-DATE
:CordovaLib:generateDebugSources UP-TO-DATE
:CordovaLib:incrementalDebugJavaCompilationSafeguard UP-TO-DATE
:CordovaLib:compileDebugJavaWithJavac UP-TO-DATE
:CordovaLib:processDebugJavaRes UP-TO-DATE
:CordovaLib:transformResourcesWithMergeJavaResForDebug UP-TO-DATE
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForDebug UP-TO-DATE
:CordovaLib:mergeDebugJniLibFolders UP-TO-DATE
:CordovaLib:transformNative_libsWithMergeJniLibsForDebug UP-TO-DATE
:CordovaLib:transformNative_libsWithSyncJniLibsForDebug UP-TO-DATE
:CordovaLib:bundleDebug UP-TO-DATE
:CordovaLib:preReleaseBuild UP-TO-DATE
:CordovaLib:checkReleaseManifest
:CordovaLib:prepareReleaseDependencies
:CordovaLib:compileReleaseAidl UP-TO-DATE
:CordovaLib:compileReleaseNdk UP-TO-DATE
:CordovaLib:copyReleaseLint UP-TO-DATE
:CordovaLib:mergeReleaseShaders UP-TO-DATE
:CordovaLib:compileReleaseShaders UP-TO-DATE
:CordovaLib:generateReleaseAssets UP-TO-DATE
:CordovaLib:mergeReleaseAssets UP-TO-DATE
:CordovaLib:mergeReleaseProguardFiles UP-TO-DATE
:CordovaLib:packageReleaseRenderscript UP-TO-DATE
:CordovaLib:compileReleaseRenderscript UP-TO-DATE
:CordovaLib:generateReleaseResValues UP-TO-DATE
:CordovaLib:generateReleaseResources UP-TO-DATE
:CordovaLib:packageReleaseResources UP-TO-DATE
:CordovaLib:processReleaseManifest UP-TO-DATE
:CordovaLib:generateReleaseBuildConfig UP-TO-DATE
:CordovaLib:processReleaseResources UP-TO-DATE
:CordovaLib:generateReleaseSources UP-TO-DATE
:CordovaLib:incrementalReleaseJavaCompilationSafeguard UP-TO-DATE
:CordovaLib:compileReleaseJavaWithJavac UP-TO-DATE
:CordovaLib:processReleaseJavaRes UP-TO-DATE
:CordovaLib:transformResourcesWithMergeJavaResForRelease UP-TO-DATE
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease UP-TO-DATE
:CordovaLib:mergeReleaseJniLibFolders UP-TO-DATE
:CordovaLib:transformNative_libsWithMergeJniLibsForRelease UP-TO-DATE
:CordovaLib:transformNative_libsWithSyncJniLibsForRelease UP-TO-DATE
:CordovaLib:bundleRelease UP-TO-DATE
:prepareComAndroidSupportMultidex101Library UP-TO-DATE
:prepareComAndroidSupportSupportCompat2531Library UP-TO-DATE
:prepareComAndroidSupportSupportCoreUi2531Library UP-TO-DATE
:prepareComAndroidSupportSupportCoreUtils2531Library UP-TO-DATE
:prepareComAndroidSupportSupportFragment2531Library UP-TO-DATE
:prepareComAndroidSupportSupportMediaCompat2531Library UP-TO-DATE
:prepareComAndroidSupportSupportV132531Library UP-TO-DATE
:prepareComAndroidSupportSupportV42531Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesAnalytics1104Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesAnalyticsImpl1104Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesBase1104Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesBasement1162Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesBasementLicense1162Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesTagmanagerV4Impl1104Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesTasks1162Library UP-TO-DATE
:prepareComGoogleAndroidGmsPlayServicesTasksLicense1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseAnalytics1104Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseAnalyticsImpl1104Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseCommon1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseCommonLicense1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseCore1104Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseIid1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseIidLicense1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseMessaging1162Library UP-TO-DATE
:prepareComGoogleFirebaseFirebaseMessagingLicense1162Library UP-TO-DATE
:prepareMeLeolinShortcutBadger1117Library UP-TO-DATE
:prepareOrgApacheCordovaCordovaLib630DebugLibrary UP-TO-DATE
:prepareDebugDependencies
:compileDebugAidl UP-TO-DATE
:compileDebugRenderscript UP-TO-DATE
:generateDebugBuildConfig UP-TO-DATE
:generateDebugResValues UP-TO-DATE
:processDebugGoogleServices
Parsing json file: D:\notification\platforms\android\google-services.json
:generateDebugResources
:mergeDebugResources UP-TO-DATE
:processDebugManifest UP-TO-DATE
:processDebugResources UP-TO-DATE
:generateDebugSources UP-TO-DATE
:incrementalDebugJavaCompilationSafeguard UP-TO-DATE
:compileDebugJavaWithJavac
:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
D:\notification\platforms\android\src\com\adobe\phonegap\push\FCMService.java:393: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
D:\notification\platforms\android\src\com\adobe\phonegap\push\FCMService.java:403: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
:compileDebugJavaWithJavac FAILED
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.001 secs
(node:11368) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: cmd: Command failed with exit code 1 Error output:
D:\notification\platforms\android\src\com\adobe\phonegap\push\FCMService.java:393: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
D:\notification\platforms\android\src\com\adobe\phonegap\push\FCMService.java:403: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
(node:11368) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by krishnaclouds** [comment]
@satsathish: Can you please specify which version of Plugin you are using? Any other version details will be helpful in reproducing the error.

**Comment by satsathish** [comment]
@krishnaclouds  these are all 
cordova-plugin-android-permissions 1.0.0 "Permissions"
cordova-plugin-appavailability 0.4.2 "AppAvailability"
cordova-plugin-barcodescanner 0.7.0 "BarcodeScanner"
cordova-plugin-camera 4.0.1 "Camera"
cordova-plugin-device 1.1.7 "Device"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-file 5.0.0 "File"
cordova-plugin-file-transfer 1.7.0 "File Transfer"
cordova-plugin-google-analytics 1.8.3 "Google Universal Analytics Plugin"
cordova-plugin-inappbrowser 1.7.2 "InAppBrowser"
cordova-plugin-network-information 1.3.4 "Network Information"
cordova-plugin-screen-orientation 3.0.1 "Screen Orientation"
cordova-plugin-splashscreen 5.0.1 "Splashscreen"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
cordova-sqlite-storage 2.1.4 "Cordova sqlite storage plugin"
es6-promise-plugin 4.2.2 "Promise"
io.hpd.cordova.gimbal2 1.0.0 "Gimbal2"
phonegap-plugin-push 2.1.3 "PushPlugin"

**Comment by macdonst** [comment]
@satsathish go into Android Studio and update your dependencies. The version of the support tools you are using is not up to date. Once you do this and it fixes your problem please come back and close this issue.

**Comment by satsathish** [comment]
After update my dependencies i got these again 

BUILD FAILED

Total time: 1 mins 36.474 secs
* What went wrong:
A problem occurred evaluating root project 'android'.
> Failed to apply plugin [id 'com.android.application']
   > Minimum supported Gradle version is 4.1. Current version is 3.3. If using the gradle wrapper, try editing the distributionUrl in D:\notification\gradle\wrapper\gradle-wrapper.properties to gradle-4.1-all.zip

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
(node:1096) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: cmd: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.

* Where:
Build file 'D:\notification\platforms\android\build.gradle' line: 20

* What went wrong:
A problem occurred evaluating root project 'android'.
> Failed to apply plugin [id 'com.android.application']
   > Minimum supported Gradle version is 4.1. Current version is 3.3. If using the gradle wrapper, try editing the distributionUrl in D:\notification\gradle\wrapper\gradle-wrapper.properties to gradle-4.1-all.zip

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
(node:1096) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by macdonst** [comment]
@satsathish the solution is right in your error message.

> Minimum supported Gradle version is 4.1. Current version is 3.3

You are using Gradle 3.3 and you need to upgrade to 4.1.

**Comment by satsathish** [comment]
@macdonst hi thank you for helping!  but i don't know how to change the version 3.3 to 4.1

i was tried tried lot of things.. i did't get succes

i change to 2.2.3 to 3.0.1
    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
    }

2.14.1 to 4.1
task wrapper(type: Wrapper) {
    gradleVersion = '2.14.1'
}

gradle/wrapper/gradle-wrapper.properties
this is my distributionUrl
distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip

i changed 4.1-all.zip and then build the application automatically it's come to 3.3

i also checked android studio -> file/project structure/project 
Gradle version is 4.1
Android Plugin Version 3.0.1

**Comment by macdonst** [comment]
@satsathish from the command line type `gradle -v`, that will tell you what version of Gradle you are using. For instance:

```
------------------------------------------------------------
Gradle 4.3.1
------------------------------------------------------------

Build time:   2017-11-08 08:59:45 UTC
Revision:     e4f4804807ef7c2829da51877861ff06e07e006d

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_144 (Oracle Corporation 25.144-b01)
OS:           Mac OS X 10.13.3 x86_64
```

**Comment by satsathish** [comment]
this is the version using in my machine

------------------------------------------------------------
Gradle 4.1
------------------------------------------------------------

Build time:   2017-08-07 14:38:48 UTC
Revision:     941559e020f6c357ebb08d5c67acdb858a3defc2

Groovy:       2.4.11
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_151 (Oracle Corporation 25.151-b12)
OS:           Windows 10 10.0 amd64

After i update to 4.1

i got these error , anyone know about this ,please let me know what is that!


Android Studio project detected
ANDROID_HOME=C:\Users\name\AppData\Local\Android\sdk
JAVA_HOME=C:\Program Files\Java\jdk1.8.0_131
studio
Subproject Path: CordovaLib
Subproject Path: app
Exception in thread "main" java.lang.NullPointerException
        at org.gradle.wrapper.PathAssembler.getDistName(PathAssembler.java:84)
        at org.gradle.wrapper.PathAssembler.getDistribution(PathAssembler.java:40)
        at org.gradle.wrapper.Install.createDist(Install.java:44)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
(node:12300) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: cmd: Command failed with exit code 1 Error output:
Exception in thread "main" java.lang.NullPointerException
        at org.gradle.wrapper.PathAssembler.getDistName(PathAssembler.java:84)
        at org.gradle.wrapper.PathAssembler.getDistribution(PathAssembler.java:40)
        at org.gradle.wrapper.Install.createDist(Install.java:44)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
(node:12300) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by macdonst** [comment]
@satsathish looks like the gradle wrapper is failing to find gradle. Please make sure that gradle is in your $PATH or considering you are on windows the %PATH% environment variable.

**Comment by satsathish** [comment]
@macdonst  how to solve this one ! i tried lot of way. i did't get success

**Comment by bhumin3i** [comment]
@satsathish Please refer link -> https://gradle.org/install/  hope it helps

**Comment by satsathish** [comment]
i tried already that also. no use! i could not build the ionic applications

Subproject Path: CordovaLib
Subproject Path: app
Exception in thread "main" java.lang.NullPointerException
        at org.gradle.wrapper.PathAssembler.getDistName(PathAssembler.java:84)
        at org.gradle.wrapper.PathAssembler.getDistribution(PathAssembler.java:40)
        at org.gradle.wrapper.Install.createDist(Install.java:44)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
(node:10676) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: cmd: Command failed with exit code 1 Error output:
Exception in thread "main" java.lang.NullPointerException
        at org.gradle.wrapper.PathAssembler.getDistName(PathAssembler.java:84)
        at org.gradle.wrapper.PathAssembler.getDistribution(PathAssembler.java:40)
        at org.gradle.wrapper.Install.createDist(Install.java:44)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
(node:10676) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by bhumin3i** [comment]
@satsathish  send log of gradle -v

**Comment by macdonst** [comment]
@satsathish if you type `gradle` from the command line and it doesn't execute then it is not in your %PATH%. You'll need to add the directory that the gradle executable is in to your windows environment variable %PATH%

**Comment by bgomessales** [comment]
Has anyone managed to solve this problem? I've been trying to solve it for weeks.

**Comment by lakshya-8** [comment]
I have been trying to fix this issue for the last one week. Tried almost everything. Did anyone find a way to fix this issue while keeping the Cordova 7.1 ?
